Common/Final.cs

```diff
@@ -0,0 +1,46 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Common
+{
+	/// <summary>
+	/// This class enforces one-time only initialization.
+	/// Normally you want to use `readonly` for that, but sometimes you can't
```

Comment:
In this PR there's code where a field is assigned to during `[OneTimeSetUp]`, which isn't a constructor. In this case this may be overkill, but I could imagine a case where such enforcement would be handy.